sklearn/preprocessing/data.py

```diff
@@ -1901,3 +1906,288 @@ def transform(self, X):
         """
         return _transform_selected(X, self._transform,
                                    self.categorical_features, copy=True)
+
+
+class QuantileNormalizer(BaseEstimator, TransformerMixin):
+    """Normalize features using quantiles information.
+
+    This Normalizer scales the features between 0 and 1, equalizing the
+    distribution of each feature to a uniform distribution. Therefore,
+    for a given feature, this normalization tends to spread out the most
+    frequent values.
+
+    The normalization is applied on each feature independently.
+    The cumulative density function of a feature is used to project the
+    original values.
+
+    Parameters
+    ----------
+    n_quantiles : int, optional (default=1000)
+        Number of quantiles to be computed. It corresponds to the number
+        of landmarks used to discretize the cumulative density function.
+
+    subsample : int, optional (default=1e5)
+        Maximum number of samples used to estimate the quantiles.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by np.random.
+
+
+    Attributes
+    ----------
+    references_ : ndarray, shape (n_quantiles,)
+        The quantiles of reference.
+
+    quantiles_ : ndarray, shape (n_quantiles, n_features)
+        The values corresponding the quantiles of reference.
+
+    f_transform_ : list of callable, shape (n_quantiles,)
+        The cumulative density function used to project the data.
+
+    f_inverse_transform_ : list of callable, shape (n_quantiles,)
+        The inverse of the cumulative density function used to project the
+        data.
+
+    See also
+    --------
+    :class:`sklearn.preprocessing.StandardScaler` to perform standardization
+    that is faster, but less robust to outliers.
+    """
+
+    def __init__(self, n_quantiles=1000, subsample=int(1e5),
+                 random_state=None):
+        self.n_quantiles = n_quantiles
+        self.subsample = subsample
+        self.random_state = random_state
+
+    def _validate_X(self, X):
+        """Private function to validate X."""
+        X = check_array(X, accept_sparse='csc')
+        # we only accept positive sparse matrix
+        if sparse.issparse(X) and X.min() < 0:
+            raise ValueError('QuantileNormalizer only accepts semi-positive'
+                             ' sparse matrices')
```

Comment:
Not "semi-positive sparse matrix" but "sparse matrices with all non-negative entries".